### PR TITLE
Makefile: build podman-remote-static with cgo disabled

### PR DESCRIPTION
**Pull request by lsm5**
Resolves: #13557

[NO NEW TESTS NEEDED]

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

@Luap99 @afbjorklund @mheon @rhatdan PTAL

/cc @containers/podman-maintainers

### CODE REVIEW

**Comment by lsm5** [comment]
binary built on rawhide worked fine on centos 9 stream.

File: Makefile [hunk 1]
```diff
@@ -336,7 +336,9 @@ $(SRCBINDIR)/podman$(BINSFX): $(SRCBINDIR) .gopathok $(SOURCES) go.mod go.sum
 		-o $@ ./cmd/podman
 
 $(SRCBINDIR)/podman-remote-static: $(SRCBINDIR) .gopathok $(SOURCES) go.mod go.sum
-	$(GOCMD) build \
+	CGO_ENABLED=0 \
+	CC=$(BINDIR)/musl-gcc \
```
Comment (Luap99) [inline Makefile:340]: Is this needed? I don't think libc is used when cgo is disabled.

Comment (lsm5) [reply Makefile:340]: let me verify without  it.

Comment (lsm5) [reply Makefile]: done thanks!

**Comment by lsm5** [comment]
Didn't we merge this too soon? I thought there was a comment from @mheon about windows, darwin needing cgo.

**Comment by Luap99** [comment]
podman-remote-static is used for linux only AFIAK, windows and macos have different makefile targets
Also I don't think you can use cgo if you compile for a different os.

**Comment by fjorgemota** [comment]
Yeah, looking here, this commit basically made `podman-remote-static` be a Windows binary, instead of a Linux binary.

Maybe `GOOS` is `windows` here for some reason?

**Pull request merged**